pkg/syncthing/syncthing.go

```diff
@@ -329,64 +332,44 @@ func (s *Syncthing) WaitForScanning(ctx context.Context, wg *sync.WaitGroup, dev
 // WaitForCompletion waits for the remote to be totally synched
 func (s *Syncthing) WaitForCompletion(ctx context.Context, wg *sync.WaitGroup, dev *model.Dev, reporter chan float64) error {
 	defer close(reporter)
-	ticker := time.NewTicker(500 * time.Millisecond)
-	folder := fmt.Sprintf("okteto-%s", dev.Name)
-	params := map[string]string{"folder": folder, "device": DefaultRemoteDeviceID}
-	completion := &Completion{}
-	var prevNeedBytes int64
-	needZeroBytesIter := 0
-	log.Infof("waiting for synchronization to complete...")
-	for {
-		select {
-		case <-ticker.C:
-			if prevNeedBytes == completion.NeedBytes {
-				if needZeroBytesIter >= 50 {
-					return errors.ErrSyncFrozen
-				}
-
-				needZeroBytesIter++
-			} else {
-				needZeroBytesIter = 0
-			}
-		case <-ctx.Done():
-			log.Debug("cancelling call to 'rest/db/completion'")
-			return ctx.Err()
-		}
-
+	params := map[string]string{
+		"events":  "FolderCompletion",
+		"device":  DefaultRemoteDeviceID,
+		"folder":  fmt.Sprintf("okteto-%s", dev.Name),
+		"timeout": "0",
+		"since":   "0",
+	}
+	var since int
+	var completion float64
+	log.Info("waiting for synchronization to complete...", completion)
+	for completion != 100 {
 		if _, err := s.APICall("rest/db/override", "POST", 200, params, true, nil); err != nil {
-			//overwrite on each iteration to avoid sude effects of remote scannings
+			//overwrite on each iteration to avoid side effects of remote scannings
 			log.Infof("error calling 'rest/db/override' syncthing API: %s", err)
 		}
-
-		prevNeedBytes = completion.NeedBytes
-		body, err := s.APICall("rest/db/completion", "GET", 200, params, true, nil)
+		body, err := s.APICall("rest/events", "GET", 200, params, true, nil)
 		if err != nil {
-			log.Infof("error calling 'rest/db/completion' syncthing API: %s", err)
+			log.Infof("error calling 'rest/events' syncthing API: %s", err)
 			continue
 		}
-		err = json.Unmarshal(body, completion)
-		if err != nil {
-			log.Infof("error unmarshaling 'rest/db/completion': %s", err)
+		var events []Event
+		if err := json.Unmarshal(body, &events); err != nil {
+			log.Infof("error unmarshalling syncthing events: %s", err)
 			continue
 		}
-
-		if completion.GlobalBytes == 0 {
-			return nil
-		}
-
-		progress := (float64(completion.GlobalBytes-completion.NeedBytes) / float64(completion.GlobalBytes)) * 100
-		log.Infof("syncthing folder is %.2f%%, needBytes %d, needDeletes %d",
-			progress,
-			completion.NeedBytes,
-			completion.NeedDeletes,
-		)
-
-		reporter <- progress
-
-		if completion.NeedBytes == 0 {
-			return nil
+		if len(events) > 0 {
```

Comment:
If the sync was completed before this code is executed, do we still get the completion events?

Reply:
Yes, we start querying from the begin of the history of the local instance:
https://github.com/okteto/okteto/pull/427/files#diff-989b94ab98dccde0ec44b8855ebd5267R340